include/bluetooth/services/ras.h

```diff
@@ -477,6 +486,51 @@ int bt_ras_rreq_rd_overwritten_subscribe(struct bt_conn *conn, bt_ras_rreq_rd_ov
  */
 int bt_ras_rreq_rd_overwritten_unsubscribe(struct bt_conn *conn);
 
+/** @brief Provide step header for each step back to the user.
+ *
+ * @param[in] subevent_header Subevent header data.
+ * @param[in] user_data       User data.
+ *
+ * @retval true If should continue parsing data.
+ *         false If data parsing should be stopped.
+ */
+typedef bool (*bt_ras_rreq_subevent_header_cb_t)(struct ras_subevent_header *subevent_header,
+						 void *user_data);
+
+/** @brief Provide step data for each step back to the user.
+ *
+ * @param[in] local_step       Local step data.
+ * @param[in] peer_step        Peer step data.
+ * @param[in] peer_step_length The length of the peer step data.
+ * @param[in] user_data        User data.
+ *
+ * @retval true If should continue parsing data.
+ *         false If data parsing should be stopped.
+ */
+typedef bool (*bt_ras_rreq_step_data_cb_t)(struct bt_le_cs_subevent_step *local_step,
+					   struct ras_rd_cs_subevent_step *peer_step,
+					   uint16_t *peer_step_length, void *user_data);
+
+/** @brief Parse peer ranging data buffer and local step data buffer.
+ *
+ * A helper for parsing ranging data-formatted buffer, as populated by @ref
+ * bt_ras_rreq_cp_get_ranging_data, and local step data buffer,
+ * where step_data_buf from le_cs_subevent_data_available has been placed together in a struct
+ * net_buf_simple.
+ *
```

Comment:
can we add some documentation here about how to use `peer_step_length` ? I think it's kind of tricky